yunwoo/programmers-2/튜플.js

```diff
@@ -0,0 +1,26 @@
+function solution(s) {
+  const result = [];
+  const nTupleCounts = {};
+  const tupleNumber = s
+    .replace(/[^0-9,]/g, '')
+    .split(',')
+    .map(Number);
+
+  tupleNumber.forEach((el) => {
+    nTupleCounts[el] = (nTupleCounts[el] || 0) + 1;
+  });
+
+  for (const count in nTupleCounts) {
+    result.push([Number(count), nTupleCounts[count]]);
+  }
+
+  result.sort((a, b) => b[1] - a[1]);
+
+  return result.map((el) => el[0]);
+}
```

Comment:
```suggestion
function solution(s) {
  const result = [];
  const nTupleCounts = {};

  s.replace(/[^0-9,]/g, "")
    .split(",")
    .forEach((el) => (nTupleCounts[el] = (nTupleCounts[el] || 0) + 1));

  for (const count in nTupleCounts) {
    result.push([+count, nTupleCounts[count]]);
  }

  result.sort((a, b) => b[1] - a[1]);

  return result.map((el) => el[0]);
}
```
- 풀이잘봤습니다! 👍
- split이후에 map으로 형변환 후에 forEach 돌릴 필요없이 바로 forEach로 진행해주셔도 이상없습니다!
- nTupleCounts에서 접두사 n의 의미는 무엇인가요..!? 😲
- 아래와 같이 메서드 체이닝으로 쭈우우욱~ 푸는 방법도 있습니다
```js
function solution(s) {
  const set = new Set();

  s.replace(/}/g, "")
    .split("{")
    .filter((el) => el)
    .map((el) =>
      el
        .split(",")
        .filter((el) => el)
        .map(Number)
    )
    .sort((a, b) => a.length - b.length)
    .map((el) => el.map((num) => set.add(num)));

  return [...set];
}
```

Reply:
- 아하 객체의 값으로 넣는건 제가 숫자를 넣는거니 굳이 형변환이 필요없군요!! 
- 문제에서 n개의 튜플 집합이라해서 넣었는데 생각해보니 굳이 넣어줄 필요가 없었던거 같습니다 😅
- 메서드체이닝 한방풀이도 살펴보겠습니다 감사합니다 👍